langchain4j-pinecone/src/main/java/dev/langchain4j/store/embedding/pinecone/PineconeEmbeddingStore.java

```diff
@@ -34,41 +37,38 @@ public class PineconeEmbeddingStore implements EmbeddingStore<TextSegment> {
 
-    private final PineconeConnection connection;
     private final String nameSpace;
     private final String metadataTextKey;
+    private final Index pineconeIndex;
+    private final Pinecone pinecone;
+    private final Consumer<List<VectorWithUnsignedIndices>> afterUpsertAction;
 
     /**
      * Creates an instance of PineconeEmbeddingStore.
      *
-     * @param apiKey      The Pinecone API key.
-     * @param environment The environment (e.g., "northamerica-northeast1-gcp").
-     * @param projectId   The ID of the project (e.g., "19a129b"). This is <b>not</b> a project name.
-     *                    The ID can be found in the Pinecone URL: https://app.pinecone.io/organizations/.../projects/...:{projectId}/indexes.
-     * @param index       The name of the index (e.g., "test").
-     * @param nameSpace   (Optional) Namespace. If not provided, "default" will be used.
+     * @param apiKey            The Pinecone API key.
+     * @param environment       The environment (e.g., "northamerica-northeast1-gcp").
+     * @param projectId         The ID of the project (e.g., "19a129b"). This is <b>not</b> a project name.
+     *                          The ID can be found in the Pinecone URL: <a href="https://app.pinecone.io/organizations/.../projects/">...</a>...:{projectId}/indexes.
+     * @param index             The name of the index (e.g., "test").
+     * @param nameSpace         (Optional) Namespace. If not provided, "default" will be used.
      * @param metadataTextKey   (Optional) The key to find the text in the metadata. If not provided, "text_segment" will be used.
+     * @param afterUpsertAction
      */
     public PineconeEmbeddingStore(String apiKey,
                                   String environment,
                                   String projectId,
                                   String index,
                                   String nameSpace,
-                                  String metadataTextKey) {
+                                  String metadataTextKey,
+                                  Consumer<List<VectorWithUnsignedIndices>> afterUpsertAction) {
 
-        PineconeClientConfig configuration = new PineconeClientConfig()
-                .withApiKey(apiKey)
-                .withEnvironment(environment)
-                .withProjectName(projectId);
 
-        PineconeClient pineconeClient = new PineconeClient(configuration);
-
-        PineconeConnectionConfig connectionConfig = new PineconeConnectionConfig()
-                .withIndexName(index);
-
-        this.connection = pineconeClient.connect(connectionConfig);
+        this.pinecone = new Pinecone.Builder(apiKey).build();
+        this.pineconeIndex = pinecone.getIndexConnection(index);
         this.nameSpace = nameSpace == null ? DEFAULT_NAMESPACE : nameSpace;
         this.metadataTextKey = metadataTextKey == null ? DEFAULT_METADATA_TEXT_KEY : metadataTextKey;
+        this.afterUpsertAction = afterUpsertAction;
```

Comment:
Could you please explain why `afterUpsertAction` is needed? Seems to not be used anywhere, also no javadoc with explanation.
I would remove it if not necessary.